crates/engine/tree/src/tree/mod.rs

```diff
@@ -1207,8 +1207,22 @@ where
                 match request {
                     EngineApiRequest::InsertExecutedBlock(block) => {
                         debug!(target: "engine::tree", block=?block.block().num_hash(), "inserting already executed block");
-                        self.state.tree_state.insert_executed(block);
+
+                        let start = Instant::now();
+
+                        self.state.tree_state.insert_executed(block.clone());
                         self.metrics.engine.inserted_already_executed_blocks.increment(1);
+
+                        let elapsed = start.elapsed();
+
+                        if self.state.tree_state.is_canonical(block.block().parent_hash) {
+                            self.emit_event(EngineApiEvent::BeaconConsensus(
+                                BeaconConsensusEngineEvent::CanonicalParentBlockAdded(
+                                    block.block().clone().into(),
+                                    elapsed,
+                                ),
+                            ));
```

Comment:
then we can simplify this.

we expect that this will always be a canonical block and can just clone the inner block value and emit this as the event

Reply:
I don't understand how to simplify this :( shouldn't I need to create a variable that checks if this block's parent extends the canonical chain and return the option?

Reply:
yeah we can keep the if check but we don't need the instant etc